app/src/main/java/ca/rmen/android/poetassistant/PoemAudioExport.java

```diff
@@ -208,5 +200,20 @@ private File getAudioFile() {
         return new File(exportFolder, TEMP_AUDIO_FILE);
     }
 
+    private final Observer<Tts.TtsState> mTtsObserver = new Observer<Tts.TtsState>() {
+        @Override
+        public void onChanged(@Nullable Tts.TtsState ttsState) {
+            if (ttsState != null
+                    && (ttsState.ttsStatus == Tts.TtsStatus.UTTERANCE_COMPLETE || ttsState.ttsStatus == Tts.TtsStatus.UTTERANCE_ERROR)
+                    && TEMP_AUDIO_FILE.equals(ttsState.utteranceId)) {
+                mHandler.post(() -> {
```

Comment:
Can remove this `mHandler.post()` as `onChanged()` should be called from the main thread.